File: app/src/main/java/social/entourage/android/api/model/feed/FeedItemAuthor.java
package social.entourage.android.api.model.feed;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

import social.entourage.android.api.model.Partner;

public class FeedItemAuthor implements Serializable {

    private static final long serialVersionUID = 3412722274231780458L;

    // ----------------------------------
    // ATTRIBUTES
    // ----------------------------------

    @SerializedName("id")
    private int userID;

    @SerializedName("display_name")
    private String userName;

    @SerializedName("avatar_url")
    private String avatarURLAsString;

    @SerializedName("partner")
    private Partner partner;

    // ----------------------------------
    // CONSTRUCTORS
    // ----------------------------------

    public FeedItemAuthor(final String avatarURLAsString, final int userID, final String userName, Partner partner) {
        this.avatarURLAsString = avatarURLAsString;
        this.userID = userID;
        this.userName = userName;
        this.partner = partner;
    }

    // ----------------------------------
    // GETTERS & SETTERS
    // ----------------------------------


    public String getAvatarURLAsString() {
        return avatarURLAsString;
    }

    public void setAvatarURLAsString(final String avatarURLAsString) {
        this.avatarURLAsString = avatarURLAsString;
    }

    public int getUserID() {
        return userID;
    }

    public void setUserID(final int userID) {
        this.userID = userID;
    }

    public String getUserName() {
        return userName == null ? "" : userName;
    }

    public void setUserName(final String userName) {
        this.userName = userName;
    }

    public Partner getPartner() {
        return partner;
    }

    public void setPartner(final Partner partner) {
        this.partner = partner;
    }

    // ----------------------------------
    // PUBLIC METHODS
    // ----------------------------------

    public boolean isSame(FeedItemAuthor author) {
        if (author == null) return false;
        if (userID != author.userID) return false;
        if (avatarURLAsString != null) {
            if (!avatarURLAsString.equals(author.avatarURLAsString)) return false;
        } else {
            if (author.avatarURLAsString != null) return false;
        }
        if (partner != null) {
            return partner.isSame(author.partner);
        } else {
            return author.partner == null;
        }
    }

}

File: app/src/main/java/social/entourage/android/api/model/feed/FeedItemAuthor.kt
package social.entourage.android.api.model.feed

import com.google.gson.annotations.SerializedName
import social.entourage.android.api.model.Partner
import java.io.Serializable
import java.util.ArrayList

class FeedItemAuthor (
    @field:SerializedName("avatar_url") var avatarURLAsString: String?,
    @field:SerializedName("id") var userID: Int,
    @field:SerializedName("display_name") var userName: String?,
    @field:SerializedName("community_roles") var communityRoles: ArrayList<String>?,
    @field:SerializedName("partner") var partner: Partner?) : Serializable {

    fun isSame(author: FeedItemAuthor?): Boolean {
        if (author == null) return false
        if (userID != author.userID) return false
        if (avatarURLAsString != null) {
            if (avatarURLAsString != author.avatarURLAsString) return false
        } else {
            if (author.avatarURLAsString != null) return false
        }
        return partner?.isSame(author.partner) ?: (author.partner == null)
    }

    companion object {
        private const val serialVersionUID = 3412733374231780458L
    }

}